pyuvdata/uvdata/uvdata.py

```diff
@@ -253,19 +261,48 @@ def __init__(self):
             "channel_width", description=desc, expected_type=float, tols=1e-3,
         )  # 1 mHz
 
-        # --- observation information ---
+        desc = (
+            "Name(s) of source(s) or field(s) observed. Type string, or if "
+            "multi_object = True, a list of strings of length (Nobjects,)."
+        )
         self._object_name = uvp.UVParameter(
-            "object_name",
-            description="Source or field " "observed (string)",
-            form="str",
-            expected_type=str,
+            "object_name", description=desc, form="str", expected_type=str,
+        )
+
+        # --- multi-object handling ---
+        desc = (
+            'Only relevant if phase_type = "phased". Specifies the that the data set '
+            "contains multiple sources within it."
+        )
+        self._multi_object = uvp.UVParameter(
+            "multi_object", description=desc, expected_type=bool, value=False,
+        )
+
+        desc = (
+            "Required if multi_object = True. Specifies the number of sources "
+            "contained within the data set."
+        )
+
+        self._Nobjects = uvp.UVParameter(
+            "Nobjects", description=desc, expected_type=int, required=False,
         )
+
+        desc = (
+            "Only relevant if multi_object = True. Dictionary containing information "
+            "individual objects, including coordinate information. Keys are matched "
+            "items in the object_name list attribute."
+        )
```

Comment:
Yup, much longer description added, as well as a pointer to the `_add_object` method documentation, since that has a much longer description of all the individual parameters.